src/internal-links/handler.js

```diff
@@ -223,6 +214,53 @@ export const opportunityAndSuggestionsStep = async (context) => {
       kpiDeltas,
     },
   );
+
+  // For suggestions already marked as FIXED, if their urlTo is no longer 404,
+  // publish any deployed fixes for those suggestions.
+  try {
+    const fixedSuggestions = await Suggestion.allByOpportunityIdAndStatus(
+      opportunity.getId(),
+      SuggestionDataAccess.STATUSES.FIXED,
+    );
+    // Prepare publishes without awaiting inside loops
+    const publishTasks = [];
+    for (const fixedSuggestion of fixedSuggestions) {
+      const urlTo = fixedSuggestion?.getData()?.urlTo;
+      if (!urlTo) {
+        // eslint-disable-next-line no-continue
+        continue;
+      }
+      // eslint-disable-next-line no-await-in-loop
+      const is404 = await isLinkInaccessible(urlTo, log);
+      if (!is404 && FixEntity && typeof FixEntity.getFixEntitiesBySuggestionId === 'function') {
+        // eslint-disable-next-line no-await-in-loop
+        const fixEntities = await FixEntity.getFixEntitiesBySuggestionId(
+          fixedSuggestion.getId(),
+        );
+        // eslint-disable-next-line no-restricted-syntax
+        for (const fixEntity of fixEntities) {
+          const deployed = FixEntity?.STATUSES?.DEPLOYED;
+          const published = FixEntity?.STATUSES?.PUBLISHED;
+          if (deployed && published && typeof fixEntity.getStatus === 'function'
+            && fixEntity.getStatus() === deployed) {
+            publishTasks.push((async () => {
+              fixEntity.setStatus(published);
+              fixEntity.setUpdatedBy?.('system');
+              await fixEntity.save();
+              log.debug(
+                `Published fix entity ${fixEntity.getId?.()} for suggestion ${fixedSuggestion.getId()}`,
+              );
+            })());
+          }
+        }
+        // eslint-disable-next-line no-await-in-loop
+        await Promise.all(publishTasks);
+      }
+    }
+  } catch (err) {
+    log.warn(`Failed to publish fix entities for FIXED suggestions: ${err.message}`);
+  }
+
```

Comment:
How about also checking for the case when the Suggestion is not in Fixed state, but the new audit finds the Suggestion as outdated and the redirect url matches our ai suggestion?